sql/expression/function/point.go

```diff
@@ -0,0 +1,143 @@
+// Copyright 2020-2021 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package function
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/dolthub/go-mysql-server/sql"
+)
+
+// Point is a function that returns a point type containing values Y and Y.
+type Point struct {
+	X sql.Expression
+	Y sql.Expression
+}
+
+var _ sql.FunctionExpression = (*Point)(nil)
+
+// NewPoint creates a new point expression.
+func NewPoint(e1, e2 sql.Expression) sql.Expression {
+	return &Point{e1, e2}
+}
+
+// FunctionName implements sql.FunctionExpression
+func (p *Point) FunctionName() string {
+	return "point"
+}
+
+// Description implements sql.FunctionExpression
+func (p *Point) Description() string {
+	return "returns a new point."
+}
+
+// Children implements the sql.Expression interface.
+func (p *Point) Children() []sql.Expression {
+	return []sql.Expression{p.X, p.Y}
+}
+
+// Resolved implements the sql.Expression interface.
+func (p *Point) Resolved() bool {
+	return p.X.Resolved() && p.Y.Resolved()
+}
+
+// IsNullable implements the sql.Expression interface.
+func (p *Point) IsNullable() bool {
+	return p.X.IsNullable() || p.Y.IsNullable()
+}
+
+// Type implements the sql.Expression interface.
+func (p *Point) Type() sql.Type {
+	return sql.PointType{}
+}
+
+func (p *Point) String() string {
+	return fmt.Sprintf("POINT(%f, %f)", p.X, p.Y)
+}
+
+// WithChildren implements the Expression interface.
+func (p *Point) WithChildren(children ...sql.Expression) (sql.Expression, error) {
+	if len(children) != 2 {
+		return nil, sql.ErrInvalidChildrenNumber.New(p, len(children), 2)
+	}
+	return NewPoint(children[0], children[1]), nil
+}
+
+func convertToFloat64(x interface{}) (float64, error) {
+	switch x := x.(type) {
+	case int:
+		return float64(x), nil
+	case int8:
+		return float64(x), nil
+	case int16:
+		return float64(x), nil
+	case int32:
+		return float64(x), nil
+	case int64:
+		return float64(x), nil
+	case uint:
+		return float64(x), nil
+	case uint8:
+		return float64(x), nil
+	case uint16:
+		return float64(x), nil
+	case uint32:
+		return float64(x), nil
+	case uint64:
+		return float64(x), nil
+	case float32:
+		return float64(x), nil
+	case float64:
+		return x, nil
+	default:
+		return 0, errors.New("point: wrong type")
```

Comment:
What does mysql do here?

```
select point("a", "b");
+------------------------------------------------------+
| point("a", "b")                                      |
+------------------------------------------------------+
| 0x00000000010100000000000000000000000000000000000000 |
+------------------------------------------------------+
```

Seems like you probably want to `Float.Convert(value)` instead of this code